File: app/src/main/java/pl/expensive/wallet/WalletsView.java
package pl.expensive.wallet;

import android.content.Context;
import android.support.annotation.MainThread;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.util.AttributeSet;
import android.widget.Toast;

import java.util.Collection;

public class WalletsView extends RecyclerView implements WalletsViewContract {
    private WalletsAdapter walletsAdapter;

    public WalletsView(Context context) {
        super(context);
        init(context);
    }

    public WalletsView(Context context, AttributeSet attrs) {
        super(context, attrs);
        init(context);
    }

    public WalletsView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        init(context);
    }

    private void init(Context context) {
        setLayoutManager(new LinearLayoutManager(context));
        setVisibility(VISIBLE);
        walletsAdapter = new WalletsAdapter();
        setAdapter(walletsAdapter);
    }

    @Override
    @MainThread
    public void showWallets(Collection<WalletViewModel> storedWallets) {
        walletsAdapter.bind(storedWallets);
    }

    @Override
    @MainThread
    public void showEmpty() {
        setVisibility(GONE);
    }

    @Override
    @MainThread
    public void showFetchError() {
        Toast.makeText(getContext(), "Cannot fetch wallets", Toast.LENGTH_SHORT).show();
    }
}

File: app/src/main/java/pl/expensive/wallet/WalletsView.kt
package pl.expensive.wallet

import android.content.Context
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.util.AttributeSet

internal class WalletsView : RecyclerView {
    private var walletsAdapter: WalletsAdapter = WalletsAdapter()

    constructor(context: Context) : super(context)
    constructor(context: Context, attrs: AttributeSet) : super(context, attrs)
    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) : super(context, attrs, defStyleAttr)

    init {
        layoutManager = LinearLayoutManager(context)
        adapter = walletsAdapter
    }

    fun update(wallets: WalletViewModel) {
        walletsAdapter.bind(listOf(wallets))
    }
}
